src/event.rs

```diff
@@ -2,400 +2,288 @@ use app::ScrollDirection;
+                        .previous_option();
                 }
-                let _ = request_redraw.try_send(());
-            }
-            KeyCode::Left => {
-                app.stocks[app.current_tab]
-                    .options
-                    .as_mut()
-                    .unwrap()
-                    .selection_mode_left();
-                let _ = request_redraw.try_send(());
             }
-            KeyCode::Right => {
-                if app.stocks[app.current_tab]
-                    .options
-                    .as_mut()
-                    .unwrap()
-                    .data()
-                    .is_some()
-                {
+        }
+        KeyCode::Down => {
+            match app.stocks[app.current_tab]
+                .options
+                .as_mut()
+                .unwrap()
+                .selection_mode
+            {
+                options::SelectionMode::Dates => {
                     app.stocks[app.current_tab]
                         .options
                         .as_mut()
                         .unwrap()
-                        .selection_mode_right();
-                    let _ = request_redraw.try_send(());
+                        .next_date();
+                }
+                options::SelectionMode::Options => {
+                    app.stocks[app.current_tab]
+                        .options
+                        .as_mut()
+                        .unwrap()
+                        .next_option();
                 }
             }
-            KeyCode::Char('?') => {
-                app.previous_mode = app.mode;
-                app.mode = app::Mode::Help;
-                let _ = request_redraw.try_send(());
+        }
+        KeyCode::Left => {
+            app.stocks[app.current_tab]
+                .options
+                .as_mut()
+                .unwrap()
+                .selection_mode_left();
+        }
+        KeyCode::Right => {
+            if app.stocks[app.current_tab]
+                .options
+                .as_mut()
+                .unwrap()
+                .data()
+                .is_some()
+            {
+                app.stocks[app.current_tab]
+                    .options
+                    .as_mut()
+                    .unwrap()
+                    .selection_mode_right();
             }
-            _ => {}
         }
-    } else if key_event.modifiers == KeyModifiers::CONTROL {
-        if let KeyCode::Char('c') = key_event.code {
+        _ => {}
+    }
+}
+
+pub fn handle_key_bindings(
+    mode: Mode,
+    key_event: KeyEvent,
+    mut app: &mut app::App,
+    request_redraw: &Sender<()>,
+) {
+    match (mode, key_event.modifiers, key_event.code) {
+        (_, KeyModifiers::CONTROL, KeyCode::Char('c')) => {
+            cleanup_terminal();
+            std::process::exit(0);
+        }
+        (Mode::AddStock, modifiers, keycode)
+            if modifiers.is_empty() || modifiers == KeyModifiers::SHIFT =>
```

Comment:
Small nitpick, I think it might be more clear if we do `if modifiers != KeyModifiers::CONTROL`?